hip_data_tools/google/adwords.py

```diff
@@ -579,3 +583,69 @@ def set_query_to_fetch_all(self, start_index: int = 0, page_size: int = 500):
                  .Limit(start_index, page_size)
                  .Build())
         self.set_query(query)
+
+
+class AdWordsReportDefinitionReader(AdWordsUtil):
+    """
+    Class to interact with the ReportDefinitionService and access fields for adwords reports
+    Args:
+        conn (GoogleAdWordsConnectionManager): Connection manager to handle the creation of
+    adwords client
+    """
+
+    def __init__(self, conn: GoogleAdWordsConnectionManager):
+        super().__init__(conn, "ReportDefinitionService", "v201809")
+
+    def get_report_fields(self, report_type: str) -> List[dict]:
+        """
+        Get a list of fields for a given report.
+        Args:
+            report_type (str): Possible reports are defined in the documentation here -
```

Comment:
if the report types are limited is it worth checking if the provided string is a valid report? Or are there too many to bother?

Reply:
too many to bother.
Plus the check is present in the underlying service. Google AdWords lib does a check and tells you in a descriptive error message.